doc/ip-interface/loopback-action/ip-interface-loopback-action-design.md

```diff
@@ -0,0 +1,318 @@
+            "mpls": "enable"            
+        }
+    }
+```
+
+### 2.2.3 Configuration migration
+No special handling is required
+
+## 2.3 Flows
+
+### 2.3.1 Loopback action set interface vlan
+#### 2.3.1.1 Handle set in runtime
+![interface vlan](images/interface_vlan.svg "Figure 2: interface vlan")
+###### Figure 2: interface vlan
+
+#### 2.3.1.2 Handle set in init
+In init, loopback action configuration is being handled as part of the IP interface creation.
+During init, all field-value pairs of each key in INTF_TABLE table (e.g. INTF_TABLE|Vlan100) are being received in intfOrch together at once.
+Those field-value pairs are the interface attributes, e.g. mac address, mpls enable, loopback action etc.
+
+### 2.3.2 Loopback action set interface Ethernet
+Loopback action set on interface Ethernet flow is the same as interface vlan flow besides config DB and app DB keys being used. Used keys are defined in [2.2.1 Config DB](#221-config-db).
+
+### 2.3.3 Loopback action set interface port-channel
+Loopback action set on interface port-channel flow is the same as interface vlan flow besides config DB and app DB keys being used. Used keys are defined in [2.2.1 Config DB](#221-config-db).
+
+### 2.3.4 IP interface creation
+If loopback action was not configured by the user, default loopback action is derived from SAI behavioral model which is forward.
+
+## 2.4 SAI
+The following will be used for setting IP interface loopback action.
+
+SAI API:
+```
+sai_router_intfs_api->set_router_interface_attribute()
+sai_router_intfs_api->create_router_interface()
+```
+
+SAI attribute:
+###### Table 2: SAI attribute
+| Loopback action  | SAI attribute                                     | Attribute value            |
+|------------------|---------------------------------------------------|----------------------------|
+| drop             | SAI_ROUTER_INTERFACE_ATTR_LOOPBACK_PACKET_ACTION  | SAI_PACKET_ACTION_DROP     |
+| forward          | SAI_ROUTER_INTERFACE_ATTR_LOOPBACK_PACKET_ACTION  | SAI_PACKET_ACTION_FORWARD  |
+
+
+```
+typedef enum _sai_router_interface_attr_t
+{
+    /**
+     * @brief Packet action when a packet ingress and gets routed on the same RIF
+     *
+     * @type sai_packet_action_t
+     * @flags CREATE_AND_SET
+     * @default SAI_PACKET_ACTION_FORWARD
+     */
+    SAI_ROUTER_INTERFACE_ATTR_LOOPBACK_PACKET_ACTION,
+
+} sai_router_interface_attr_t;
+```
+
+## 2.5 Statistics
+Packets that will dropped due to loopback action will be counted in RX_ERR in IP interface statistics.
+```
+root@sonic:~# show interfaces counters rif
+      IFACE    RX_OK    RX_BPS    RX_PPS    RX_ERR    TX_OK    TX_BPS    TX_PPS    TX_ERR
+-----------  -------  --------  --------  --------  -------  --------  --------  --------
+Ethernet232        0  0.00 B/s    0.00/s         0        0  0.00 B/s    0.00/s         0
+```
+
+## 2.6 CLI
+### 2.6.1 Config command
+
+The following command will be added to configure IP interface loopback action:
```

Comment:
these commands will be created based on the auto generated tool. right?

Reply:
Unfortunately no. The command I am adding is under “config interface“ tree.
If we want to have auto generated CLI for the new command, we must auto generated CLI for the whole “config interface“ tree.
We can not have part of the tree auto generated and part of the tree not.